tools/analysis/diagnostic.go

```diff
@@ -28,11 +28,23 @@ type SuggestedFix = errors.SuggestedFix[ast.TextEdit]
 
 type TextEdit = ast.TextEdit
 
+type Severity uint8
+
+const (
+	SeverityUnknown Severity = iota
+	SeverityInfo
+	SeverityError
+	SeverityWarning
+)
+
 type Diagnostic struct {
 	Location         common.Location
 	Category         string
+	Severity         Severity
```

Comment:
The severity was intentionally omitted, [like it is in Go's `analysis.Diagnostic`](https://pkg.go.dev/golang.org/x/tools/go/analysis#Diagnostic) (which this package is heavily inspired by)

> The Diagnostic struct does not have a field to indicate its severity because opinions about the relative importance of Analyzers and their diagnostics vary widely among users. The design of this framework does not hold each Analyzer responsible for identifying the severity of its diagnostics. Instead, we expect that drivers will allow the user to customize the filtering and prioritization of diagnostics based on the producing Analyzer and optional Category, according to the user's preferences.

Could we keep this design decision, or does the lack of this field cause problems in packages using this package (e.g. the linter, language server, etc.)?

Reply:
Gotcha, the idea was to make all diagnostics in https://github.com/onflow/cadence-tools/pull/285 as severe as they are obviously necessary changes.  However, I can also do this through the category field